[UnifiedPDF] Snapshots after the first page are partially dark-shaded

sammygill opened the pull request:
#### 3f7f7ed16b4fb150585c2adba8d3913f72ba1712
<pre>
[UnifiedPDF] Snapshots after the first page are partially dark-shaded
<a href="https://bugs.webkit.org/show_bug.cgi?id=273213">https://bugs.webkit.org/show_bug.cgi?id=273213</a>
<a href="https://rdar.apple.com/126817702">rdar://126817702</a>

Reviewed by Abrar Rahman Protyasha and Simon Fraser.

277273@main attempted to address this issue by making adding in the
white background as a fill as part of the async rendering codepath. It
turns out that this was quite not enough and snapshots of pages after
the first page are still appearing as dark. It looks like this is due
to the fact that we modify the graphics context by making it so that the
origin is at the bottom of the page before we pass it off to PDFKit.
So, instead we will start adding the fill before we perform this
coordinate space transformation.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::paintPDFContent):

Canonical link: <a href="https://commits.webkit.org/277965@main">https://commits.webkit.org/277965@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7ab9d45a96e5fbcf22626fedbce225e35f2b7ca9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49033 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28250 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52002 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51740 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45103 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34226 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25782 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40077 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50246 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25892 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42269 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21185 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23344 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7136 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45279 "Passed tests") | [⏳ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/macOS-Monterey-Release-WK2-Tests-EWS "Waiting to run tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53654 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24089 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-AppleSilicon-Sonoma-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47393 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25368 "Built successfully") | [⏳ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/macOS-Release-WK2-Stress-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46364 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10797 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26160 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25087 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
Alternatively, we could fill a translated + flipped `pageDestinationRect`? That way there aren't two `if (!asyncRenderer)` conditions, which looks a little awkward.

sammygill committed:
[UnifiedPDF] Snapshots after the first page are partially dark-shaded
https://bugs.webkit.org/show_bug.cgi?id=273213
rdar://126817702

Reviewed by Abrar Rahman Protyasha and Simon Fraser.

277273@main attempted to address this issue by making adding in the
white background as a fill as part of the async rendering codepath. It
turns out that this was quite not enough and snapshots of pages after
the first page are still appearing as dark. It looks like this is due
to the fact that we modify the graphics context by making it so that the
origin is at the bottom of the page before we pass it off to PDFKit.
So, instead we will start adding the fill before we perform this
coordinate space transformation.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::paintPDFContent):

Canonical link: https://commits.webkit.org/277965@main
